### PR TITLE
v1.0.7 Release Updates

### PR DESCRIPTION
**Pull request by Gazza7205**
## v1.07 Updates
- Support added for running the Gateway without [Diskless Config](#diskless-configuration)
  - Uses node.properties which can be mounted via [Secret or Secret Store CSI Driver](https://secrets-store-csi-driver.sigs.k8s.io/)
  - Must be conciously enabled (limited to Gateway v11.1.1)
- Redis configuration update
  - Additional system properties for the key/value store assertion added (commented by default)
    - please refer to [Techdocs](https://techdocs.broadcom.com/us/en/ca-enterprise-software/layer7-api-management/api-gateway/congw11-1/policy-assertions/assertion-palette/service-availability-assertions/key-value-storage-assertions.html#_c8b71b7b-dd84-4ee6-9771-d0bc262c36e9_sys_prop_configs) for more details
  - Using new shared state provider config **(limited to Redis and Gateway v11.1.1)**
    - this new configuration is **not backwards or forwards compatible**
      - Please view [redis configuration](https://github.com/CAAPIM/layer7-operator/blob/main/docs/gateway.md#gatewayspecappredis) for more details on how to configure your values file.
    - additional redis providers can be set via additionalConfigs
 - Configurable Java Min/Max Heap size
   - Java Min and Max Heap Size is now [configurable](https://github.com/CAAPIM/layer7-operator/blob/main/docs/gateway.md#gatewayspecappjavajvmheap)
 - Liquibase Log Level is now settable via database.liquibaseLogLevel.
   - default "off"
     - possible values
       - severe
       - warning
       - info
       - fine(debug)
       - off
 - System Properties
   - FIPS
     - Switched to BCFIPS 2.0 to provide both non-FIPS and FIPS functionality to the Gateway.
      - Previous
        - com.safelogic.cryptocomply.rsa.allow_multi_use=true
      - New
        - com.l7tech.org.bouncycastle.rsa.allow_multi_use=true
 - Port Restart on Key Change
   - Added spec.app.listenPorts.refreshOnKeyChanges to listenPort definition
   - This adds refreshOnKeyChanges=true to advanced properties on 8443 and 9443 when  spec.app.listenPorts.harden is true
   - New system-property for refreshing mtls key in the route via http(s) assertion
     - com.l7tech.server.policy.assertion.ServerHttpRoutingAssertion.refreshOnKeyChanges (true|false ==> default: false)
   - Fixed NodePort on the Gateway Services 
   - Graphman Client Update for Gateway v11.1.1
     - Supports the latest graphman schema
     - Supports variable expansion for entities separated into files with --options.level 1 and 2
       - Trusted Certificates
       - Keys
       - SOAP services (wsdl)
     - New entities
       - Audit Configurations
       - Roles
       - Generic Entities
     - Supports Bundle Mappings
     - Updated error handling

### CODE REVIEW

File: charts/layer7-operator/crds/gateway-crd.yaml [hunk 1]
```diff
@@ -2048,39 +1689,51 @@ spec:
...
-                                description: "insecureEdgeTerminationPolicy indicates
-                                  the desired behavior for insecure connections to
-                                  a route. While each router may make its own decisions
-                                  on which ports to expose, this is normally port
-                                  80. \n * Allow - traffic is sent to the server on
-                                  the insecure port (default) * Disable - no traffic
-                                  is allowed on the insecure port. * Redirect - clients
-                                  are redirected to the secure port."
+                                description: insecureEdgeTerminationPolicy indicates
+                                  the desired behavior for insecure c
```
Comment (uppoju) [inline charts/layer7-operator/crds/gateway-crd.yaml:1711]: incomplete sentence, is this expected ?

File: charts/layer7-operator/crds/gateway-crd.yaml [hunk 2]
```diff
@@ -2090,76 +1743,36 @@ spec:
...
-                                matches only a single label. You cannot have a wildcard
-                                label by itself (e.g. Host == \"*\"). Requests will
-                                be matched against the Host field in the following
-                                way: 1. If host is precise, the request matches this
-                                rule if the http host header is equal to Host. 2.
-                                If host is a wildcard, then the request matches this
-                                rule if the http host header is to equal to the suffix
-                                (removing the first label) of the wildcard rule."
+                              description: host is the fully qualified domain name
+                                of a network host, as defined by RF
```
Comment (uppoju) [inline charts/layer7-operator/crds/gateway-crd.yaml:1750]: here also

File: charts/layer7-operator/crds/l7api-crd.yaml [hunk 3]
```diff
@@ -3,7 +3,7 @@ kind: CustomResourceDefinition
 metadata:
   annotations:
```
Comment (uppoju) [inline charts/layer7-operator/crds/l7api-crd.yaml:4]: description is truncated in crd files

**Review by uppoju — APPROVED** [review]

File: charts/layer7-operator/crds/gateway-crd.yaml (hunk 1 above)
Comment (Gazza7205) [reply charts/layer7-operator/crds/gateway-crd.yaml:1711]: Yes - set the maxDescLength to 75. If this isn't done, the CRDs become too large to install.

```
.PHONY: manifests
manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
#$(CONTROLLER_GEN) rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
	$(CONTROLLER_GEN) rbac:roleName=manager-role crd:generateEmbeddedObjectMeta=true,maxDescLen=75 webhook paths="./..." output:crd:artifacts:config=config/crd/bases
```

Comment (Gazza7205) [reply charts/layer7-operator/crds/gateway-crd.yaml:1711]: The /docs are not limited by this and contain the full description

File: charts/layer7-operator/crds/gateway-crd.yaml (hunk 2 above)
Comment (Gazza7205) [reply charts/layer7-operator/crds/gateway-crd.yaml:1750]: See above

**Pull request merged**